avm/src/vm/deploy.rs

```diff
@@ -0,0 +1,185 @@
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Debug, Serialize)]
+struct AWSCredentials {
+  accessKeyId: String,
+  secretAccessKey: String,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+struct AWSConfig {
+  credentials: AWSCredentials,
+  region: String,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+struct Config {
+  aws: AWSConfig,
+}
+
+#[derive(Deserialize, Debug)]
+struct App {
+  id: String,
+  url: String,
+  status: String,
+}
+
+const CONFIG_NAME: &str = ".alan/deploy.json";
+const CONFIG_SCHEMA: &str = "Please define one with the following schema: \n{
+  \"aws\": {
+      \"region\": \"string\",
+      \"credentials\": {
+        \"accessKeyId\": \"string\",
+        \"secretAccessKey\": \"string\",
+      }
+  }
+}";
+const HOW_TO_AWS: &str = "
+To create an AWS access key follow this tutorial:\n\nhttps://aws.amazon.com/premiumsupport/knowledge-center/create-access-key/\n\n
+Then enable programmatic access for the IAM user, and attach the built-in 'AdministratorAccess' policy to your IAM user.
+";
+
+fn get_config() -> Config {
+  let home = std::env::var("HOME").unwrap();
+  let file_name = &format!("{}/{}", home, CONFIG_NAME);
+  let path = Path::new(file_name);
+  let file = File::open(path);
+  if file.is_err() {
+    println!("No deploy config at {}", file_name);
+    println!("{}", CONFIG_SCHEMA);
+    println!("{}", HOW_TO_AWS);
+    std::process::exit(1);
+  }
+  let reader = BufReader::new(file.unwrap());
+  let config = from_reader(reader);
+  if config.is_err() {
+    println!("Invalid deployment config. Error: {}", config.unwrap_err());
+    println!("{}", CONFIG_SCHEMA);
+    println!("{}", HOW_TO_AWS);
+    std::process::exit(1);
+  }
+  config.unwrap()
+}
+
+fn post(url: String, body: Value) -> Result<String, Box<dyn Error>> {
+  let client = Client::builder().build::<_, Body>(hyper_tls::HttpsConnector::new());
+  let rt  = Runtime::new().unwrap();
+  rt.block_on(async move {
+    let req = Request::post(url)
+      .header("Content-Type", "application/json")
+      .body(body.to_string().into())?;
+    let mut resp = client.request(req).await?;
+    let data = hyper::body::to_bytes(resp.body_mut()).await?;
+    let str = String::from_utf8(data.to_vec())?;
+    Ok(str)
+  })
+}
+
+fn get_app_str(agz_file: &str) -> String {
+  let path = Path::new(agz_file);
+  if path.extension().is_none() || path.extension().unwrap() != "agz" {
+    println!("Deploy failed. The provided file must be an .agz file");
+    std::process::exit(1);
+  }
+  let app = read(agz_file).expect(&format!("Deploy failed parsing {}", agz_file));
+  return base64::encode(app);
+}
+
+pub fn kill(app_id: &str) {
+  let body = json!({
+    "deployConfig": get_config(),
+    "appId": app_id,
+  });
+  let res = post(format!("{}/v1/kill", URL), body);
+  if res.is_ok() {
+    println!("Killing app with id {} if it exists...\n", app_id);
+    status();
+  } else {
+    println!("Killing app with id {} failed.", app_id);
+  }
```

Comment:
maybe `match` the result of the `post` so we can better report the error?

Reply:
the correct way would be to print the error from the REST API, but AWS never fails the terminate instance request so this branch should never fail/run. Down the line, the endpoint could do another request to the AWS sdk and check if it worked, but this is okay as-is for now